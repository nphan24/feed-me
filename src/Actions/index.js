export const postRecipes = recipes => ({
  type: 'POST_RECIPES',
  recipes
});

export const replaceRecipes = recipes => ({
  type: 'REPLACE_RECIPES',
  recipes
});

export const addFavorite = recipe => ({
  type: 'ADD_FAVORITE',
  recipe
});

export const removeFavorite = id => ({
  type: 'REMOVE_FAVORITE',
  id
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const promptSignIn = error => ({
  type: 'PROMPT_SIGNIN',
  error
});