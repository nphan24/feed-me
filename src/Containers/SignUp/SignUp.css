.signup-component {
  background-color: #e2dcdc;
  width: 500px;
  margin: auto auto;
  text-align: center;
  box-shadow: -10px 5px 15px grey;
}

.signup-comment {
  margin: 0;
  padding-top: 50px;
  font-size: 2.5em;
  font-weight: 400;
}

.signup-field {
  display: block;
  margin: auto;
  width: 300px;
  height: 40px;
  margin-top: 15px;
  border-radius: 15px;
  padding-left: 15px;
  padding-top: 5px;
  font-size: 36px;
}

.signup-field::-webkit-input-placeholder {
  font-size: 36px;
}

.bottom-input {
  margin-bottom: 15px;
}

.signup-submit-button {
  margin-bottom: 15px 0 15px 0;
  width: 120px;
  height: 40px;
  border-radius: 15px;
  font-size: 1em;
  color: black;
}

.signup-submit-button:hover {
  background-color : #abd8ea;
}

.login-link {
  font-size: 1.5em;
  text-decoration: none;
}

.login-link:hover {
  color:#BA55D3;
}

.signup-error {
  font-size: 2em;
  color: red;
}

