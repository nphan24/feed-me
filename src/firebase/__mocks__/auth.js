export const login = jest.fn().mockImplementation(() => [{
  email: 'dog@gmail.com', 
  password: 'dogdog', 
  uid: 24}
]);