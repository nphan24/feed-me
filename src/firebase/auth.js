import { auth } from './firebase';

//sign up
export const signUp = (email, password) => 
  auth.createUserWithEmailAndPassword(email, password);

  //login
export const login = (email, password) =>
  auth.signInWithEmailAndPassword(email, password);

  //signout
export const doSignOut = () => auth.signOut();