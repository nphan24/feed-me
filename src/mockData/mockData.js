/* eslint-disable */

export const mockRecipe = {
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
};

export const favorites = {
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
};

export const favoritesArray = [{
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
}];

export const uncleanedMockRecipeData = {
  matches: [
    {
      recipeName: 'tofu',
      imageUrlsBySize: 'url',
      ingredients: ['', ''],
      totalTimeInSeconds: 3600,
      id: 2,
      pasta: 'speghetti',
      fruit: 'mango'
    }
  ]
};

export const cleanedMockRecipeData = [
  {
    Ingredients: ['', ''],
    id: 2,
    image: undefined,
    name: 'tofu',
    totaltime: 60
  }
];

export const mockRecipesArray = [
  {
    name: 'pasta',
    image: 'url',
    totalTime: '30',
    id: 2
  },
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];

export const mockRemainingArray = [
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];

export const mockUser = {
  username: 'dog',
  email: 'dog@gmail.com',
  password: 'dogdog'
};

export const mockSignupUser = {
  username: '',
  email: 'a@gmail.com',
  passwordOne: 'aaaaaa',
  passwordTwo: 'aaaaaa',
  error: false
}

export const mockUserArray = [
  {
    username: 'dog',
    email: 'dog@gmail.com',
    password: 'dogdog'
  }
];

export const returnedRecipes = {
  attributes: {
    course: ['Side Dishes']
  },
  attribution: {
    html:
      "<a href='http://www.yummly.co/recipe/Potatoes-Au-Gâ€¦ly' src='https://static.yummly.co/api-logo.png'/>",
    url: 'http://www.yummly.co/recipe/Potatoes-Au-Gratin-2244864',
    text: 'Potatoes Au Gratin recipes: information powered by Yummly',
    logo: 'https://static.yummly.co/api-logo.png'
  },
  cookTime: '1 Hr',
  cookTimeInSeconds: 3600,
  flavors: { Piquant: 0 },
  id: 'Potatoes-Au-Gratin-2244864',
  images: [{}],
  ingredientLines: [
    'Butter to grease the baking dish',
    '2-1/2 cups heavy cream',
    '1-1/2 teaspoons salt',
    '1/4 teaspoon ground black pepper',
    '3 large Russet or Yukon Gold potatoes (about 2-1/4 pounds), peeled and sliced very thin',
    '1 cup grated Parmigiano-Reggiano'
  ],
  name: 'Potatoes Au Gratin',
  numberOfServings: 6,
  prepTime: '15 Min',
  prepTimeInSeconds: 900,
  rating: 4,
  source: {
    sourceDisplayName: 'Once Upon A Chef',
    sourceSiteUrl: 'http://www.onceuponachef.com',
    sourceRecipeUrl:
      'https://www.onceuponachef.com/recipes/potatoes-au-gratin.html'
  },
  totalTime: '1 hr 15 min',
  totalTimeInSeconds: 4500,
  yield: '6'
};

export const recipes = [
  {
    Ingredients: [
      'butter',
      'heavy cream',
      'salt',
      'ground black pepper',
      'russet potatoes',
      'parmigiano'
    ],
    id: 'Potatoes-Au-Gratin-2244864',
    image:
      'https://lh3.googleusercontent.com/1UaqgvAZrw346jUpat4-at6mkz6-Jhjymn8AkVY2Kwu48E0TXbuH9OnhsR_k5OF-5UV_MSxx6qbnFPnawzTtqQ=s90-c',
    name: 'Potatoes Au Gratin',
    source: 'https://www.onceuponachef.com/recipes/potatoes-au-gratin.html',
    totaltime: 75
  }
];
