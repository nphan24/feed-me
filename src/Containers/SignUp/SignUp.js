import React, {Component} from 'react';
import { connect } from 'react-redux';
import * as routes from '../../constants/routes';
import { NavLink, withRouter } from 'react-router-dom';
import { auth, db } from '../../firebase';
import * as Actions from '../../Actions';
import PropTypes from 'prop-types';
import './SignUp.css';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = { 
      username: '', 
      email: '', 
      passwordOne: '', 
      passwordTwo: '', 
      error: false 
    };
  }

  handleInput = (event) => {
    const { name, value } = event.target;

    this.setState({ [name]: value });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const {
      username,
      email,
      passwordOne
    } = this.state;

    try {
      const authUser = await auth.signUp(email, passwordOne);

      await db.doCreateUser(authUser.uid, username, email);
      const user = {
        username,
        email: authUser.email,
        uid: authUser.uid
      };

      this.props.addUser(user);
      this.setState({
        username: '',
        email: '',
        passwordOne: '',
        passwordTwo: '',
        error: false
      });
      this.props.history.push('/');
    } catch (error) {
      this.setState({ error });
    }
  }
  
  render() {
    const {
      username,
      email,
      passwordOne,
      passwordTwo,
      error
    } = this.state;

    const isInvalid = passwordOne !== passwordTwo || 
    passwordOne === '' || 
    email === '' || 
    username === '';
    
    return (
      <div className='signup-component'>
        <form className="signup-form" onSubmit={this.handleSubmit}>
          <h2 className='signup-comment'>Create New Account</h2>
          <label>
            <input 
              type="text" 
              placeholder="username"
              className='signup-field' 
              value={username} 
              name="username" 
              onChange={this.handleInput} 
            />
            <input 
              type="email" 
              placeholder="Email"
              className='signup-field'
              value={email} 
              name="email" 
              onChange={this.handleInput} 
            />
            <input 
              type="password" 
              placeholder="Password"
              className='signup-field' 
              value={passwordOne} 
              name="passwordOne" 
              onChange={this.handleInput} 
            />
            <input
              type="password" 
              placeholder="Confirm Password"
              className='signup-field bottom-input' 
              value={passwordTwo} 
              name="passwordTwo" 
              onChange={this.handleInput} 
            />
            <button disabled={isInvalid} 
              type="submit" 
              className='signup-submit-button'>Sign Up
            </button>
          </label>
        </form>
        <NavLink to={routes.LOGIN} className="login-link">
          Have an account? Login!
        </NavLink>
        {error && <p className='signup-error'>{error.message}</p>}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(Actions.addUser(user))
});

SignUp.propTypes = {
  user: PropTypes.object,
  addUser: PropTypes.func,
  history: PropTypes.object
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SignUp));
