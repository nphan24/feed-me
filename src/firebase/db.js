import { db } from './firebase';

export const doCreateUser = (uid, username, email) => {
  db.ref(`users/${uid}`).set({
    username,
    email,
    uid
  });
};

export const onceGetUsers = () => 
  db.ref('users').once('value');

export const addFavoriteToDB = (uid, favorite) => {
  db.ref(`favorites/${uid}`).set(
    favorite
  );
};

export const removeFavoriteFromDB = async (uid, favoriteId) => {
  const userFavorites = await getFavorites(uid);
  delete userFavorites[favoriteId];
  addFavoriteToDB(uid, userFavorites);
};

export const getFavorites = (uid) => {
  const favorites = db.ref(`favorites/${uid}`).once('value');
  return favorites;
};