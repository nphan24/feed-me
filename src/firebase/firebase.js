import * as firebase from 'firebase';

const config = {
  apiKey: 'AIzaSyCaGyiQQKcQbzZgiCyHLLHIGOvX5E4VrH0',
  authDomain: 'feed-me-2be60.firebaseapp.com',
  databaseURL: 'https://feed-me-2be60.firebaseio.com',
  projectId: 'feed-me-2be60',
  storageBucket: 'feed-me-2be60.appspot.com',
  messagingSenderId: '713865205266'
};

firebase.initializeApp(config);

const auth = firebase.auth();
const database = firebase.database();

export { auth };
