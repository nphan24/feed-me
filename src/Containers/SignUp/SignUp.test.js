import { SignUp, mapStateToProps, mapDispatchToProps } from '../SignUp/SignUp';
import React from 'react';
import { shallow } from 'enzyme';
import * as mock from '../../mockData/mockData';
import * as Actions from '../../Actions';
import * as firebase from '../../firebase/auth';

describe('SignUp', () => {
  let wrapper;
  let mockUser = mock.mockUser;
  let mockaddUser = jest.fn();
  let mockHistory = {push: jest.fn()};

  beforeEach(()=> {
    wrapper = shallow(<SignUp
      user = {mockUser}
      addUser = {mockaddUser}
      history = {mockHistory}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state on handleInput', () => {
    const mockEvent = { target: { value: 'dog@gmail', name: 'dog' } };
    wrapper.instance().handleInput(mockEvent);
    expect(wrapper.state('dog')).toEqual('dog@gmail');
  });

  it('should call signUp on handlesubmit', () => {
    const event = { preventDefault: jest.fn() };
    
    firebase.signUp = jest.fn().mockImplementation(()=> Promise.resolve({
      email: 'dog@gmail.com', 
      uid: 24
    }));
    wrapper.instance().handleSubmit(event);
    expect(firebase.signUp).toHaveBeenCalled();
  });

  it('should call addUser in handleSubmit', () => {
    const event = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(event);
    const user = { email: 'dog@gmail.com', uid: 24, username: '' };

    expect(mockaddUser).toHaveBeenCalledWith(user);
  });

  it('should set state back to its default values', () => {
    const event = { preventDefault: jest.fn() };
    const expected = '';

    wrapper.setState(mock.mockSignupUser);
    wrapper.instance().handleSubmit(event);
    expect(wrapper.state('username')).toEqual(expected);
  });

  it('should call history.push with the correct params', () => {
    const event = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(event);
    expect(mockHistory.push).toHaveBeenCalledWith('/');
  });

  it('should set the state to the error message if firebase fails to retreive a user', async () => {
    const event = { preventDefault: jest.fn() };
    firebase.signUp = jest.fn().mockImplementation(() => Promise.reject('error')
    );
    await wrapper.instance().handleSubmit(event);
    await expect(wrapper.state('error')).toEqual('error');
  });
});

describe('mapStateToProps', () => {
  it('correctly maps users to props ', () => {
    const user = mock.mockUser;
    const expected = user;
    const mockState = { user };
    const mapped = mapStateToProps(mockState);
    expect(mapped.user).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with correct params on addUser', () => {
    const mockDispatch = jest.fn();
    const user = mock.mockUser;
    const expected = Actions.addUser(user);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addUser(user);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});