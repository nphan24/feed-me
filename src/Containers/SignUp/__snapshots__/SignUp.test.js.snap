// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    addUser={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    user={
        Object {
            "email": "dog@gmail.com",
            "password": "dogdog",
            "username": "dog",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="signup-form"
          onSubmit={[Function]}
>
          <h2
                    className="signup-comment"
          >
                    Create New Account
          </h2>
          <label>
                    <input
                              className="signup-field"
                              name="username"
                              onChange={[Function]}
                              placeholder="username"
                              type="text"
                              value=""
                    />
                    <input
                              className="signup-field"
                              name="email"
                              onChange={[Function]}
                              placeholder="Email"
                              type="email"
                              value=""
                    />
                    <input
                              className="signup-field"
                              name="passwordOne"
                              onChange={[Function]}
                              placeholder="Password"
                              type="password"
                              value=""
                    />
                    <input
                              className="signup-field bottom-input"
                              name="passwordTwo"
                              onChange={[Function]}
                              placeholder="Confirm Password"
                              type="password"
                              value=""
                    />
                    <button
                              className="signup-submit-button"
                              disabled={true}
                              type="submit"
                    >
                              Sign Up
                    </button>
          </label>
</form>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="login-link"
          to="/login"
>
          Have an account? Login!
</NavLink>,
        false,
      ],
      "className": "signup-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="signup-comment"
>
              Create New Account
</h2>,
            <label>
              <input
                            className="signup-field"
                            name="username"
                            onChange={[Function]}
                            placeholder="username"
                            type="text"
                            value=""
              />
              <input
                            className="signup-field"
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            type="email"
                            value=""
              />
              <input
                            className="signup-field"
                            name="passwordOne"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
              />
              <input
                            className="signup-field bottom-input"
                            name="passwordTwo"
                            onChange={[Function]}
                            placeholder="Confirm Password"
                            type="password"
                            value=""
              />
              <button
                            className="signup-submit-button"
                            disabled={true}
                            type="submit"
              >
                            Sign Up
              </button>
</label>,
          ],
          "className": "signup-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create New Account",
              "className": "signup-comment",
            },
            "ref": null,
            "rendered": "Create New Account",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="signup-field"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value=""
/>,
                <input
                  className="signup-field"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
/>,
                <input
                  className="signup-field"
                  name="passwordOne"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
/>,
                <input
                  className="signup-field bottom-input"
                  name="passwordTwo"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  type="password"
                  value=""
/>,
                <button
                  className="signup-submit-button"
                  disabled={true}
                  type="submit"
>
                  Sign Up
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-field",
                  "name": "username",
                  "onChange": [Function],
                  "placeholder": "username",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-field",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-field",
                  "name": "passwordOne",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signup-field bottom-input",
                  "name": "passwordTwo",
                  "onChange": [Function],
                  "placeholder": "Confirm Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                  "className": "signup-submit-button",
                  "disabled": true,
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "button",
              },
            ],
            "type": "label",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Have an account? Login!",
          "className": "login-link",
          "to": "/login",
        },
        "ref": null,
        "rendered": "Have an account? Login!",
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="signup-form"
            onSubmit={[Function]}
>
            <h2
                        className="signup-comment"
            >
                        Create New Account
            </h2>
            <label>
                        <input
                                    className="signup-field"
                                    name="username"
                                    onChange={[Function]}
                                    placeholder="username"
                                    type="text"
                                    value=""
                        />
                        <input
                                    className="signup-field"
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="Email"
                                    type="email"
                                    value=""
                        />
                        <input
                                    className="signup-field"
                                    name="passwordOne"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                        />
                        <input
                                    className="signup-field bottom-input"
                                    name="passwordTwo"
                                    onChange={[Function]}
                                    placeholder="Confirm Password"
                                    type="password"
                                    value=""
                        />
                        <button
                                    className="signup-submit-button"
                                    disabled={true}
                                    type="submit"
                        >
                                    Sign Up
                        </button>
            </label>
</form>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="login-link"
            to="/login"
>
            Have an account? Login!
</NavLink>,
          false,
        ],
        "className": "signup-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="signup-comment"
>
                Create New Account
</h2>,
              <label>
                <input
                                className="signup-field"
                                name="username"
                                onChange={[Function]}
                                placeholder="username"
                                type="text"
                                value=""
                />
                <input
                                className="signup-field"
                                name="email"
                                onChange={[Function]}
                                placeholder="Email"
                                type="email"
                                value=""
                />
                <input
                                className="signup-field"
                                name="passwordOne"
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                                value=""
                />
                <input
                                className="signup-field bottom-input"
                                name="passwordTwo"
                                onChange={[Function]}
                                placeholder="Confirm Password"
                                type="password"
                                value=""
                />
                <button
                                className="signup-submit-button"
                                disabled={true}
                                type="submit"
                >
                                Sign Up
                </button>
</label>,
            ],
            "className": "signup-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create New Account",
                "className": "signup-comment",
              },
              "ref": null,
              "rendered": "Create New Account",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="signup-field"
                    name="username"
                    onChange={[Function]}
                    placeholder="username"
                    type="text"
                    value=""
/>,
                  <input
                    className="signup-field"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
/>,
                  <input
                    className="signup-field"
                    name="passwordOne"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
/>,
                  <input
                    className="signup-field bottom-input"
                    name="passwordTwo"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
/>,
                  <button
                    className="signup-submit-button"
                    disabled={true}
                    type="submit"
>
                    Sign Up
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-field",
                    "name": "username",
                    "onChange": [Function],
                    "placeholder": "username",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-field",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "Email",
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-field",
                    "name": "passwordOne",
                    "onChange": [Function],
                    "placeholder": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signup-field bottom-input",
                    "name": "passwordTwo",
                    "onChange": [Function],
                    "placeholder": "Confirm Password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up",
                    "className": "signup-submit-button",
                    "disabled": true,
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": "button",
                },
              ],
              "type": "label",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Have an account? Login!",
            "className": "login-link",
            "to": "/login",
          },
          "ref": null,
          "rendered": "Have an account? Login!",
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
